diff --git a/python/tvm/tir/buffer.py b/python/tvm/tir/buffer.py
index ec57ad780..f0b4e3208 100644
--- a/python/tvm/tir/buffer.py
+++ b/python/tvm/tir/buffer.py
@@ -185,7 +185,10 @@ class Buffer(Object, Scriptable):
         if not isinstance(indices, (tuple, list)):
             indices = [indices]
         has_slice = any(isinstance(i, slice) for i in indices)
-        has_step = any(isinstance(i, slice) and i.step is not None for i in indices)
+        has_step = any(
+            isinstance(i, slice) and (i.step is not None and i.step != 1)
+            for i in indices
+        )
         analyzer = Analyzer()
         if has_slice and not has_step:
             region = []
